tools/code-generation/smithy/codegen/cpp-smoke-tests-codegen/src/main/java/com/amazonaws/util/awsclientsmithygenerator/generators/SmokeTestsParser.java

```diff
@@ -0,0 +1,362 @@
+                        String key = paramEntry.getKey();
+                        Shape fieldShape =  fieldShapeMap.get(key);
+                        if (fieldShape == null) {
+                            continue;
+                        }
+                        
+                        Node value = paramEntry.getValue();
+                        try {
+                            blockWriter.addCode(String.format("input.Set%s(%s);\n", GenericCodegenAdapter.capitalizeFirstLetter(key),
+
+                                codegenAdapter.GenerateCppSetters(
+                                        test.getOperationName().toLowerCase() + "_elem",
+                                        value,
+                                        fieldShape, //useful for C++ return type object
+                                        0, //useful for depth
+                                        0, //useful for array elements at same depth
+                                        functionMap,
+                                        functionOrder)
+                                ));
+                        } catch (Exception e) {
+                            throw new RuntimeException(e);
+                        }
+                    }
+                    
+                    //test.setRequestBlock(sb.toString());
+                    test.setRequestBlock(blockWriter.getCode());
+
+                    List<String> lines = new ArrayList<>();
+
+                    // Iterate through each value in the map
+                    for (String functionName : functionOrder) 
+                    {
+                        String value = functionMap.get(functionName);
+                        
+                        if(value != null)
+                        {
+                            // Split the value by newline (\n)
+                            String[] splitLines = value.split("\n");
+
+                            // Add each line to the vector
+                            lines.addAll(Arrays.asList(splitLines));
+                        }
+                    }
+
+                    test.setTestDataCodeBlock(lines);
+                }
+            }
+
+            //get expectations
+            if(testcase.getExpectation().isSuccess())
+            {
+                test.setExpectSuccess(true);
+            }
+            else
+            {
+                test.setExpectSuccess(false);
+                if(testcase.getExpectation().getFailure().isPresent() &&
+                    testcase.getExpectation().getFailure().get().getErrorId().isPresent())
+                {
+                    test.setErrorShapeId(Optional.of(testcase.getExpectation().getFailure().get().getErrorId().get().getName()) );
+                }
+            }
+
+            //get configuration properties
+            ClientConfiguration config = new ClientConfiguration();
```

Comment:
should create a static function in ClientConfiguration called `FromSmokeModelTest`. that can be called

```
ClientConfiguration::FromSmokeModelTest(AwsSmokeTestModel).getOrElseThrow(new WahteverException(...))
```

then move this logic into that static function, since config really isnt optional we can throw a exception.